### ISSUE TITLE
Issue please help / advise.

### ISSUE THREAD

**Issue by spidernope** [Nandaka/DanbooruDownloader#200]
Every imageboard works fine except one.

Rule34hentai.net won't let me view content on the main tab or bulk download.

I get the following error when I seach a tag on the main tab.

![Danbooru Downloader 20181029 Error](https://user-images.githubusercontent.com/39782214/74082575-a17ca000-4a5b-11ea-9a29-e7689e4c3365.png)

**Comment by Nandaka** [comment]
yep, checked the data retrieved from the url and it returns cloudflare page.

**Comment by spidernope** [comment]
Is it possible to create a fix? Something similar to the Pixiv Downloader.

I remember adding some 'code-string' to the "session_id" in some data-file.

This allowed users to bypass the captcha login measure Pixiv added.

**Comment by Nandaka** [comment]
You can try this:
1. Open the website using browser (I'm using Vivaldi), ensure the Cloudflare check is completed, and then get the cookie value from Developer Tools 
 by pressing F12 and navigate to Network tab (you might need to refresh the page).
2. Copy the cookie value and user agent. ![image](https://user-images.githubusercontent.com/836867/74317299-1cc0b780-4db6-11ea-801a-4ff90b8cdd6f.png)
3. Using Notepad, edit the `DanbooruProviderList.xml` and paste the whole cookie value to the Username section. Ensure the LoginType = Cookie.
![image](https://user-images.githubusercontent.com/836867/74317392-4da0ec80-4db6-11ea-9161-01803c02125d.png)
4. Using Notepad, edit `DanbooruDownloader3.exe.config` and update the user agent based on your browser value. 
![image](https://user-images.githubusercontent.com/836867/74317529-8e990100-4db6-11ea-8b11-d4417a22eb0d.png)

**Comment by spidernope** [comment]
Sorry to bother you again but no luck, I copied the entire string but still get an error.

I installed the Vivaldi browser, but my string looks different.

Yours seems to start with shm_user, mine does not.

Which parts do I need and which parts can I ignore.

Also in my developer mode, I can not find the 'index' page to view the cookie.

**Comment by Nandaka** [comment]
You need to be logged in to the site. This is the page that I use on that page: 
`https://rule34hentai.net/index.php?q=/rss/images/3D/1`

**Comment by spidernope** [comment]
Yes, I'am logged in. Can you compare your cookie to mine? Which parts I need in which order.

**Comment by azzxxa** [comment]
i tryed but can't work what should i do?

**Comment by spidernope** [comment]
Did some testing and made some discoveries
1. The CloudFlare check creates a session, this session is valid for 1hr max.
2. This session is only valid as long as you web browser remains open.
My conclusion is, you browser must remain open and after 1 hour you need to re-import your cookie string.

@azzxxa - Do you want me to write you a step by step guide on what to do?

**Comment by pencilbook** [comment]
@spidernope yes please

**Comment by pencilbook** [comment]
I installed Vivaldi, did the same as @Nandaka, but I am getting crashes when I select the website in "Provider" and search for a tag.

**Comment by Nandaka** [comment]
most likely you paste the cookie value incorrectly in `DanbooruProviderList.xml`, make sure you copy the whole value and ensure the closing bracket is not accidentaly deleted in the xml.

**Comment by pencilbook** [comment]
I think it's because I don't have log4net and HtmlAgilityPack installed. Do you have a download link for log4net 1.2.11 (newkey) and HtmlAgilityPack because I can only find the download link for 2.0.8 on logging.apache.org and I don't know how to install HtmlAgilityPack from nuget.org. 
Windows 10 here btw

**Comment by spidernope** [comment]
Apologies that I did not reply earlier. My PC would not work over the weekend, TLDR my GPU died.
I'm currently wrting a step-by-step guide. The following bit is what I suspect most people did wrong.

1. Open DanbooruProviderList.xml
2. Scroll down to line 282
3. Change "<UserName />" into "<UserName>
4. Add your whole cookie string
5. Change LoginType to "Cookie"

I suspect most people forgot to add the 'closeing' bracket. Added a screenshot, left side is unedited and the right side is what it should look like.

![Cookie](https://user-images.githubusercontent.com/39782214/74759066-7f172d80-5278-11ea-9cd9-f0c0cc3395e5.png)

**Comment by spidernope** [comment]
This is my step by step guide.

I used the new Microsoft Edge (chromium based) browser to extract my cookie string.
I also used NotePad++ to edit the code.

1. Download Browser from https://www.microsoft.com/en-us/edge
2. Go to rule34hentai.net
3. Wait for the CloudFlare check to complete
4. Login with your account
5. Browse to the main page https://rule34hentai.net/post/list
6. Press F12 to enable developer mode
7. Press “Crtl + R” to load website data
![Screenshot 1](https://user-images.githubusercontent.com/39782214/74762952-226b4100-527f-11ea-976c-8d38b5c3231c.png)
8. Click on “Type” then on “List”
![Screenshot 2](https://user-images.githubusercontent.com/39782214/74762971-27c88b80-527f-11ea-8017-131c37616b75.png)
9.  Under the header “Request Headers” copy your cookie
![Screenshot 3](https://user-images.githubusercontent.com/39782214/74762990-2c8d3f80-527f-11ea-813b-ee9a7b791b84.png)
10. Paste the entire cookie string into “DanbooruProviderList.xml” on line 282
11. Safe and close the file
12. Under the header “Request Headers” copy the user-agent
13. Paste the user-agent string into “DanbooruDownloader3.exe.config” on line 222
14. Safe and close the file

Now DanbooruDownloader3 should work once again on rule34hentai.net
These are the steps I took to fix the CloudFlare protection.

Remember:
- The CloudFlare check creates a session, this session is valid for 1hr max
- This session is only valid as long as your webbrowser remains open
- My conclusion is, your browser must remain open and after 1 hour you need to re-import your cookie string

**Comment by pencilbook** [comment]
I'm getting this error when I search for something https://i.imgur.com/3ld2DaO.png

**Comment by spidernope** [comment]
I would also get that error from time to time.
-Do you keep your web browser open?
-Check you imported your cookie string correctly
-Change the opening and closing bracket
-Change the login type
-Updated the useragent string
The step by step guide is the only solution I can offer. I know this is of no help but that is how I fixed it.

**Comment by pencilbook** [comment]
I don't have log4net 1.2.11 and HtmlAgilityPack. Did you install those packages, and how? I can't find it on the log4net's website and I don't know how to install HtmlAgilityPack because they don't have an option for Windows

**Comment by Nandaka** [comment]
log4net: https://logging.apache.org/log4net/download_log4net.cgi get the latest version should be ok.
HtmlAgilityPack: https://www.nuget.org/packages/HtmlAgilityPack/ use nuget to install the package from your visual studio.

**Comment by lee27** [comment]
sorry but I don't know what this cloudfire is

**Comment by spidernope** [comment]
@lee27 CloudFlare is DDOS and Spam protection for websites and services.

**Comment by kyummii** [comment]
Thank you very much :))
